### PR TITLE
use spacing and typography lib

### LINKED ISSUE #522: Styling issues caused by withWhiteSpace HOC
In attempting to update the styles for list components to better match govuk-frontend, it's been discovered that using the `withWhiteSpace` HOC can be problematic, causing unintended consequences when attempting to set styles for child components.

Take this example of a very simplified version of the `OrderedList` component:
```
const OrderedList = styled('ol')(
  {
    marginTop: 0,
    // Lists within a list should get a marginTop
    [`& &, ${UnorderedList}`]: {
      marginTop: 10,
    },
  }
)
```

This component forms part of a family of 3 list components together with `UnorderedList` and `ListItem`.

When using Emotion with the above code, `ListItem` components get a `marginTop: 10` set.  The reason for this is that as all 3 list components are wrapped with the `withWhiteSpace` HOC, the class name value for the `UnorderedList` selector above is identical to a class name being applied to `ListItem` in the DOM - it is effectively a `withWhiteSpace` selector.

With styled-components, the logical way to implement `UnorderedList` is to just define it as `OrderedList.withComponent('ul')`, and thus the selector in the example above becomes just `& &`.  Unfortunately when using nested `UnorderedList`, whilst the `marginTop` rule ends up being applied, but other styles for `UnorderedList` do *not*.  This again appears to be down to wrapping with the `withWhiteSpace` HOC.

In both cases the simple fix is to remove the `withWhiteSpace` HOC wrapping and just use the styled components directly.  This is not desirable however as it removes support for adjusting margins and padding.

It is thus suggested that the `withWhiteSpace` HOC be deprecated/retired, and replaced with a styling function generator that can provide the same support.

This would be used thus:
```
const Component = styled('div')(
  { ...some styles here },
  withWhiteSpace(configAsPerExistingHOC)
);
```

### PR DESCRIPTION
**Pull request by stevesims**
most `withWhiteSpace` HOC wrapping now done with spacing lib

bug fixes to styling in `Details` component
  * marginBottom wasn't configured correctly
  * `:hover` had extraneous `:` preventing it from working

Some remaining font styling replaced with calls to typography.font

Fixes #522

**Checklist**:

* [x] Documentation
* [ ] Tests N/A
* [x] Ready to be merged

### CODE REVIEW

**Comment by penx** [comment]
Deploy preview for *govuk-react* processing.

Building with commit cf736749947df3f1ea795fa8da9f66c4ba75817c

https://app.netlify.com/sites/govuk-react/deploys/5c6d47ae3a525900098d7f75

**Comment by penx** [comment]
Deploy preview for *govuk-react* processing.

Building with commit 2a475be00707acde7b273dccf1c2c484bcaa3830

https://app.netlify.com/sites/govuk-react/deploys/5c6d47f3c3a720000a082105

**Comment by stevesims** [comment]
Improved standardisation of whitespace and typography has meant lots of small visual diffs, mostly with fixed line-spacing

File: components/details/src/index.js [hunk 1]
```diff
@@ -92,8 +92,8 @@ const DetailsText = styled('div')({
  * - https://github.com/alphagov/govuk-frontend/blob/master/src/components/details/_details.scss
  */
 const Details = ({ summary, children, ...props }) => (
-  <StyledDetails {...props}>
-    <StyledSummary><SummaryText>{summary}</SummaryText></StyledSummary>
+  <StyledDetails {...props} role="group">
```
Comment (penx) [inline components/details/src/index.js]: please review this bit

**Pull request merged**